frontend/appflowy_flutter/lib/plugins/database/application/row/related_row_detail_bloc.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:appflowy/workspace/application/view/view_service.dart';
+import 'package:appflowy_backend/dispatch/dispatch.dart';
+import 'package:appflowy_backend/protobuf/flowy-database2/protobuf.dart';
+import 'package:bloc/bloc.dart';
+import 'package:collection/collection.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+import '../database_controller.dart';
+import 'row_controller.dart';
+
+part 'related_row_detail_bloc.freezed.dart';
+
+class RelatedRowDetailPageBloc
+    extends Bloc<RelatedRowDetailPageEvent, RelatedRowDetailPageState> {
+  RelatedRowDetailPageBloc({
+    required String databaseId,
+    required String initialRowId,
+  }) : super(const RelatedRowDetailPageState.loading()) {
+    _dispatch();
+    _init(databaseId, initialRowId);
+  }
+
+  @override
+  Future<void> close() {
+    state.when(
+      loading: () {},
+      ready: (databaseController, rowController) {
+        rowController.dispose();
+        databaseController.dispose();
+      },
+    );
+    return super.close();
+  }
+
+  void _dispatch() {
+    on<RelatedRowDetailPageEvent>((event, emit) async {
+      event.when(
+        didInitialize: (databaseController, rowController) {
+          state.maybeWhen(
+            ready: (_, oldRowController) {
+              oldRowController.dispose();
+              emit(
+                RelatedRowDetailPageState.ready(
+                  databaseController: databaseController,
+                  rowController: rowController,
+                ),
+              );
+            },
+            orElse: () {
+              emit(
+                RelatedRowDetailPageState.ready(
+                  databaseController: databaseController,
+                  rowController: rowController,
+                ),
+              );
+            },
+          );
+        },
+      );
+    });
+  }
+
+  void _init(String databaseId, String initialRowId) async {
+    final getDatabaseResult = await DatabaseEventGetDatabases().send();
```

Comment:
By the way, I have added fold support to the asynchronous function.

You can use `await DatabaseEventGetDatabases().send().fold(...)`;